zeppelin-web/src/app/notebook/paragraph/paragraph.controller.js

```diff
@@ -830,6 +830,29 @@ function ParagraphCtrl ($scope, $rootScope, $route, $window, $routeParams, $loca
       // autocomplete on 'ctrl+.'
       $scope.editor.commands.bindKey('ctrl-.', 'startAutocomplete')
 
+      // Show autocomplete on tab
+      $scope.editor.commands.addCommand({
+        name: 'tabAutocomplete',
+        bindKey: {
+          win: 'tab',
+          mac: 'tab',
+          sender: 'editor|cli'
+        },
+        exec: function(env, args, request) {
+          let iColumnPosition = $scope.editor.selection.getCursor().column
+
+          // If user has pressed tab on first line char or if editor mode is %md, keep existing behavior
+          // If user has pressed tab anywhere in between and editor mode is not %md, show autocomplete
+          if (iColumnPosition && $scope.paragraph.config.editorMode !== 'ace/mode/markdown') {
```

Comment:
Thanks @malayhm for an opinion. Actually i'm on the exact same page what you're talking.

Immutable editor configuration I mentioned above is

  - Applied per interpreter. So mixed paragraph type will have different autocomplete shortcuts. This is a major benefit of having shortcut configuration in editor configuration.
  - User can not change editor configuration inside of interpreter-setting.json. It is fixed in compile time. So we don't have to worry about changing autocomplete shortcut until Zeppelin has shortcuts customize feature.

What do you think? do you have any other concerns?